### PR TITLE
feat(components/List): Disable item selection and disable "select all"

### PR DESCRIPTION
**Pull request by yyanwang**
**What is the problem this PR is trying to solve?**
A TMC feature needs to disable certain items in a list, as well as the "select all" checkbox. 
This PR adds the ability to disable the selection of checkboxes for list items and also "select all" checkbox.
Additionally, a new story named `List with selection disabled items` has been added.

**What is the chosen solution to this problem?**
The `getRowState` function is used for a list item to check if its state.disabled property is true, which indicates that selection is disabled for that item.
`isToggleAllDisabled` is a function that can be passed to the select all checkbox. If the function returns true, the select all checkbox will be disabled. Conversely, if it returns false, the select all checkbox will be enabled.

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by Gbacc — APPROVED** [review]

**Review by jmainguytalend — COMMENTED** [review]
- Please take care of Eslint warns.

- Is it possible to remove pointer cursor on disabled checkbox ?

File: packages/components/src/VirtualizedList/CellCheckbox/CellCheckbox.test.js [hunk 1]
```diff
@@ -28,6 +28,20 @@ describe('CellActions', () => {
 		expect(wrapper.getElement()).toMatchSnapshot();
 	});
 
+	it('should render disabled checkbox', () => {
```
[the remark this answers is not in the document]
Comment (yyanwang) [reply packages/components/src/VirtualizedList/CellCheckbox/CellCheckbox.test.js:32]: Rewrite this test case in RTL.

File: packages/components/src/VirtualizedList/HeaderCheckbox/HeaderCheckbox.test.js [hunk 2]
```diff
@@ -44,6 +44,17 @@ describe('Header "Select All" checkbox', () => {
 		expect(checkbox.prop('disabled')).toBe(true);
 	});
 
+	it('should render disabled checkbox when isToggleAllDisabled() is true', () => {
```
[the remark this answers is not in the document]
Comment (yyanwang) [reply packages/components/src/VirtualizedList/HeaderCheckbox/HeaderCheckbox.test.js:49]: Rewrite in RTL too.

File: packages/components/src/List/ListComposition.stories.js [hunk 3]
```diff
@@ -698,6 +698,45 @@ export const SelectableItems = () => {
...
 
+export const SelectableItemsWithDisabledItems = () => {
+	const { isSelected, onToggleAll, onToggleItem } = List.hooks.useCollectionSelection(
+		simpleCollection,
+		[],
+		'id',
+	);
+
+	const getRowState = item => (item.id === 1 ? { disabled: true } : {});
+	const isToggleAllDisabled = () => {
```
[the remark this answers is not in the document]
Comment (yyanwang) [reply packages/components/src/List/ListComposition.stories.js]: Changed to use parameter.

**Comment by yyanwang** [comment]
Hello @jmainguytalend 
Thanks for review.
Fixed all eslint errors and use `cursor: not-allowed` for disabled checkbox.

File: packages/components/src/VirtualizedList/utils/tablerow.test.js [hunk 4]
```diff
@@ -23,8 +27,8 @@ describe('tablerow', () => {
 		it('should NOT insert selection column when selection callback is NOT provided', () => {
 			// given
 			const children = [
-				<VirtualizedList.Content label="Id" dataKey="id" width={50} />,
-				<VirtualizedList.Content label="Name" dataKey="name" width={350} />,
+				<VirtualizedList.Content key={0} label="Id" dataKey="id" width={50} />,
+				<VirtualizedList.Content key={1} label="Name" dataKey="name" width={350} />,
```
Comment (yyanwang) [inline packages/components/src/VirtualizedList/utils/tablerow.test.js:31]: added key to get rid of eslint warnings

**Review by jmainguytalend — APPROVED** [review]

**Pull request merged**